packages/crypto/src/crypto.ts

```diff
@@ -48,6 +54,73 @@ export const getPublicKey = (
   return p256.getPublicKey(privateKey, isCompressed);
 };
 
+/**
+ * HPKE Encrypt Function
+ * Encrypts data using Hybrid Public Key Encryption (HPKE) standard https://datatracker.ietf.org/doc/rfc9180/.
+ *
+ * @param {HpkeEncryptParams} params - The encryption parameters including plain text, encapsulated key, and sender private key.
+ * @returns {Uint8Array} - The encrypted data.
+ */
+
+export const hpkeEncrypt = ({
+  plainText,
+  encappedKeyBuf,
+  senderPriv,
+}: HpkeEncryptParams): Uint8Array => {
+  try {
+    const senderPubBuf = getPublicKey(
+      uint8ArrayFromHexString(senderPriv),
+      false
+    );
+    const aad = buildAdditionalAssociatedData(senderPubBuf, encappedKeyBuf);
+    // Step 1: Convert the plain text string to a Uint8Array
+    const plaintextBuf = new TextEncoder().encode(plainText);
```

Comment:
We shouldn't make the assumption that the encrypted content is utf8 text. It's not true in many cases. Let's take a uin8array as an argument here, it'll be way more flexible (otherwise there's no way to encrypt a raw random secret here since not all bytes will be valid utf8 characters)

Reply:
On it 🫡